tests/integration/packet/packet.exp

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env -S expect -f
+#
+# This is an expect script modeling the user interaction of PATE on the packet example.
+#
+# Beyond testing basic verifier functionality, this test is also intended to
+# provide an integration test of the 1) terminal user interaction, and 2)
+# resulting equivalence condition, in ways that the other automated tests do not.
+# The test for the "expected" equivalence condition is syntactic, so this test will
+# likely be more brittle (e.g. break if the term simplification changes such that
+# intermediate variable numbers change) and may need to be updated to reflect
+# intended subsequent changes to PATE verification details.
+#
+# This test is intended to be run from the root of the pate directory:
+#    ./tests/integration/packet/packet.exp
+#
+# Note that expected strings must be escaped: double-quotes (\") and brackets (\\\[)
+
+spawn ./pate.sh --original tests/integration/packet/exe/packet.exe --patched tests/integration/packet/exe/packet.patched.exe -s parse_packet
+
+set timeout 480
+
+expect_before {
+        timeout { send_user "\n\nFAILURE: timeout\n"; exit 1 }
+        eof     { send_user "\n\nFAILURE: eof\n";     exit 2 }
+}
+
+expect "Choose Entry Point"
+expect "1: Function Entry \"parse_packet\" (segment1+0x554)"
+expect "?>"
+send "1\r"
+
+expect "Handle observable difference:"
+expect "4: Avoid difference with equivalence condition"
+expect "?>"
+send "4\r"
+
+expect "Continue verification?"
+expect "0: Finish and view final result"
+expect "?>"
+send "0\r"
+
+expect "15: Final Result"
+expect ">"
+send "15\r"
+
+expect "0: Assumed Equivalence Conditions"
+expect "2: Binaries are conditionally, observably equivalent"
+expect ">"
+send "0\r"
+
+# Inspect the specific equivalence condition
+
+expect "0: segment1+0x644 \\\[ via: \"parse_packet\" (segment1+0x554) \\\]"
+expect ">"
+send "0\r"
+
+expect "3: let -- segment1+0x664.. in not v40591"
+expect ">"
+send "3\r"
+
+expect "v40591 = and (eq 0x80:\\\[8\\\] v40584) (not (eq v40587 (bvSum v40584 0x80:\\\[8\\\])))"
```

Comment:
We probably want to figure out how to avoid making this depend on the exact variable identifiers (e.g. v40591) since those will almost certainly change fairly regularly.

Reply:
If we pulled the simplification logic from the GUI into Pate, would that fix this issue? I believe want to do that anyway as time permits.

Reply:
Agreed - this will be relatively brittle for now so we'll have to replace this line when changes to the verifier result in changes to this expression. Ultimately we'd like to match not on this low-level view of the equivalence condition expression but instead on the simplified version we show to operators, so we could match something like:

```
(#x80 != read1#00011045)) | (read1#00011044) = (#x80 + read1#x00011045)))
```

but currently the simplification logic for producing this expression spans both the verifier and the Binary Ninja plugin. I've opened issue #420 about unifying this logic in the verifier, once that is resolved it will natural to update our regression test to leverage it.